### ISSUE TITLE
Add except: ["blockless-after-same-name-blockless"] option to at-rule-empty-line-before

### ISSUE THREAD

**Issue by jaydenseric** [stylelint/stylelint#1366]
This config seems buggy:

```json
{
  "rules": {
    "at-rule-empty-line-before": ["always", {
      "except": ["blockless-group" ]
    }]
  }
}
```

With this CSS it will unexpectedly result in an error:

```css
@import '../foo';

@custom-media --breakpoint-1 (); /* ← Unexpected empty line before at-rule */
@custom-media --breakpoint-2 ();
```

Reordering different atrule groups can also cause an error.

This has been an issue for some time and affects stylelint version v6.5.1.

Docs: [`at-rule-empty-line-before`](http://stylelint.io/user-guide/rules/at-rule-empty-line-before).

**Comment by evilebottnawi** [comment]
@jaydenseric Thanks.
/cc @davidtheclark @jeddy3 should we ignore `custom-media` in `at-rule-*` rules?

**Comment by davidtheclark** [comment]
@jaydenseric : I think it's actually not "buggy". You said with your config that groups of blockless at-rules should never have empty lines before them — and that's what you're receiving the warning about.

Would this problem be solved by adding a `ignore: ["blockless-group"]` option (rather than just an `except` one)?

**Comment by jeddy3** [comment]
> Would this problem be solved by adding a ignore: ["blockless-group"] option (rather than just an except one)?

Yep, I think it would. However, I think there's an option that would allow @jaydenseric to *enforce* (rather than just *ignore*) his chosen convention: `except: ["matching-name-blockless-group"]`.

Which would enforce this:

```css
@import '../foo';
@import '../bar';

@custom-media --breakpoint-1 (); /* ← Empty line between different types of at-rule blockless groups */
@custom-media --breakpoint-2 ();
```

Whereas `except: ["blockless-group"]` would enforce this:

```css
@import '../foo';
@import '../bar';
@custom-media --breakpoint-1 (); /* ← no empty line within a blockless group */
@custom-media --breakpoint-2 ();
```

Is there a better name for the option than `"matching-name-blockless-group"`?

@jaydenseric Is this something you'd like to contribute? It should be a fairly simple addition as the new option's code will be very similar to that of the `except: ["blockless-group"]` option.

**Comment by davidtheclark** [comment]
> except: ["matching-name-blockless-group"]

That'd be good, too.

**Comment by jaydenseric** [comment]
I'm confused, it looks like this has been prematurely closed? The commits add `ignore: ["blockless-after-same-name-blockless"]` without `except: ["blockless-after-same-name-blockless"]`.

I was originally confused what "blockless-group" meant, thinking it meant same-name at-rule groups for some reason. https://github.com/stylelint/stylelint/issues/1366#issuecomment-223789231 describes the requirements well.

**Comment by jeddy3** [comment]
> I'm confused, it looks like this has been prematurely closed? 

Yes, you're right. Reopening.
